plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2975,10 +2971,7 @@ bool producer_plugin_impl::push_read_only_transaction(transaction_metadata_ptr t
    try {
       auto start = fc::time_point::now();
       chain::controller& chain = chain_plug->chain();
-      if ( !chain.is_building_block() ) {
-         _ro_exhausted_trx_queue.push_front( {std::move(trx), std::move(next)} );
-         return true;
-      }
+      EOS_ASSERT( chain.is_building_block(), producer_exception, "Expecting chain is building block" );
```

Comment:
I don't think this is correct. We allow read-only trxs on a producer node. Those trxs could run when we are not building a block. I think this change should be reverted.

Reply:
Oh now single thread and multi-threaded resad-only trxs use the same method. For `--read-only-threads` enabled nodes, will the assert be applicable?

Reply:
Not for our test case were we allow that with producer configured. I think best to just have it gracefully handle it by pushing it back into the exhausted queue. i.e. revert this change.